### DISCUSSION TITLE
Investigate using a third-party CBR/CBZ viewer

### DISCUSSION THREAD

**Discussion by devnoname120** [devnoname120/epubviewer#15] [Ideas] [answered]
Some ideas:
- https://github.com/afzafri/Web-Comic-Reader
- ~~https://github.com/codedread/kthoom~~ (ugly, slow, and buggy)

Or I may drop comic support altogether because I don't really care about it

**Comment by roru69** [comment]
I can only speak for myself. I hope that comic support stays. I am on Nextcloud 28 with the epubviewer activated. I use it a lot.
This forked version still supports the latest version of Nextcloud without any issues (at least for me).

Regards Roru69

**Comment by devnoname120** [comment]
@roru69 Can you try [kthoom](https://codedread.com/kthoom/index.html) and report back?
I would like to have your feedback on whether it's a suitable replacement for the current CBR/CBZ implementation or not.

**Comment by roru69** [comment]
@devnoname120 I have tried kthoom and i had a bit of succes and failure.

The current epubviewer (provided by you) opens all my Comic files without any problems. Kthoom opens about 20% of the files. At this moment I don't know why Kthoom doesn't open all CBR files. Maybe it has something to do with the compression method.

It needs some more investigation. I will do so when i have some more time. After the holidays ;-)

Regards, Roru69

**Comment by devnoname120** [comment]
@roru69 Thanks a lot for your feedback. Could you open issues on their repository? I would like to see if it's just a one-off bug or if their implementation is completely broken and unreliable.

More questions: Is the UX/UI of kthoom good or bad? User-friendly or not? Is opening CBR/CBZ fast or slow in your experience?
I'm interested in your feedback on both desktop and mobile

**Comment by devnoname120** [comment]
Another alternative: https://github.com/afzafri/Web-Comic-Reader

@roru69 Could you test it and report back? It's available here: https://afzafri.github.io/Web-Comic-Reader/

**Comment by roru69** [comment]
"Another alternative: https://github.com/afzafri/Web-Comic-Reader"

That looks promising. Every CBR/CBZ file i threw at it opened without any issue. It looks nice, loadingtimes are fast. No complaints. ;-)

"Or I may drop comic support altogether because I don't really care about it"

I hope you don't do that. I use nextcloud in combination with the epub-viewer a lot to read Comics.

Roru69

**Comment by thenamelessthing** [comment]
I hope you will not drop the CBR/CBZ files. In fact I hope you improve the support of compressed CBR/CBZ files. "finger crossed" ;-)

**Reply by devnoname120** [reply]
@thenamelessthing Btw feel free to leave a rating on the Nextcloud app store now that we have RAR 5 support :) The minimum for star average to display 5 reviews and we have only 4 reviews! So currently it shows a “neutral” rating which is 3/5 stars (a bit stupid but heh that’s how the Nextcloud app store works…)

https://apps.nextcloud.com/apps/epubviewer#comments

**Reply by thenamelessthing** [reply]
Done ✅

**Comment by devnoname120** [comment, marked as answer]
For the time being I decided to fix the current CBZ/CBR viewer instead of switching to another one.

The last three releases improved the state of things quite a bit already.

Changelogs:
- [`1.9.5`](https://github.com/devnoname120/epubviewer/releases/tag/1.9.5)
- [`1.9.6`](https://github.com/devnoname120/epubviewer/releases/tag/1.9.6)
- [`1.9.7`](https://github.com/devnoname120/epubviewer/releases/tag/1.9.7)
